packages/marketplace/src/components/pages/developer-edition-download/download-modal.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import { Modal, Button } from '@reapit/elements'
+
+interface DeveloperEditionDownloadProps {
+  handleDownload: () => void
+}
+
+export const DeveloperEditionDownload: React.FC<DeveloperEditionDownloadProps> = ({ handleDownload }) => {
+  return (
+    <Modal visible title="Agency Cloud Developer Edition">
+      <>
+        <p>
+          To start the download, please click ‘Start Download’ below. It will automatically download an application file
+          which you will need to run and install the required document files. For more information or support using the
+          Developer Edition, please <a>click here</a>
+        </p>
+        <Button className="is-pulled-right mt-5" onClick={handleDownload}>
+          START DOWNLOAD
+        </Button>
+      </>
+    </Modal>
+  )
+}
+
+export const DeveloperDownloadCompleted: React.FC = () => {
+  return (
+    <Modal visible title="Download Complete">
+      <p>The download of the Developer Edition of Agency Cloud has been completed.</p>
+    </Modal>
+  )
+}
+
+export const onDownload = (setIsCompleted: React.Dispatch<boolean>) => (): void => {
+  setIsCompleted(true)
+  window.open(window.reapit.config.developerEditionDownloadUrl, '_self')
+}
+
+export const DeveloperEdition: React.FC = () => {
+  const [isCompleted, setIsCompleted] = useState<boolean>(false)
+  const handleDownload = onDownload(setIsCompleted)
+
+  if (isCompleted) return <DeveloperDownloadCompleted />
```

Comment:
Our modal will be removed from the DOM tree when visible=false, so the markup here should be just like this for simplicity, no need to create a function component for it

```
<Modal visible={isCompleted} title="Download Complete">
   <p>The download of the Developer Edition of Agency Cloud has been completed.</p>
</Modal>
```

Reply:
thanks @nphivu414. I moved those components into one.